### ISSUE TITLE
Support *args without dispatch

### ISSUE THREAD

**Issue by ellisonbg** [mrocklin/multipledispatch#2]
One way of handling `*args` would be to simple not do matching on that part. This would allow defining n-ary operators in terms of the binary version in the following form:

```
@dispatch(Expr, Expr)
def add(x, y, *args):
    result = add(x,y)
    for i in range(len(args)):
        result = add(result, args[i])
    return begin
```

**Comment by hameerabbasi** [comment]
I just ran into needing this for something. [`np.tensordot`](https://docs.scipy.org/doc/numpy/reference/generated/numpy.tensordot.html) has an `axes` kwarg. Now anyone using our "overload" would need to pass in something explicitly as a kwarg, even though the original function supports posargs.

**Comment by cpcloud** [comment]
@llllllllll Can this be closed in light of #89? This issue is referring to a slightly different problem than the one that #89 solves, but I believe you can achieve what you want with a variadic union type (or `object`), no?

**Comment by llllllllll** [comment]
Yeah, you can just tack [object] on the end of the param list for this now.
